pkg/apitype/events.go

```diff
@@ -40,9 +40,20 @@ type DiagnosticEvent struct {
 	Message string `json:"message"`
 	Color   string `json:"color"`
 	// Severity is one of "info", "info#err", "warning", or "error".
-	Severity  string `json:"severity"`
-	StreamID  int    `json:"streamID,omitempty"`
-	Ephemeral bool   `json:"ephemeral,omitempty"`
+	Severity        string           `json:"severity"`
+	StreamID        int              `json:"streamID,omitempty"`
+	Ephemeral       bool             `json:"ephemeral,omitempty"`
+	PolicyViolation *PolicyViolation `json:"policyViolation,omitempty"`
+}
+
+// PolicyViolation represents the information related to a Policy violation.
+type PolicyViolation struct {
+	// Identifier is uniquely identified a Policy. It is a Policy Pack name and version as
+	// well as the Policy name encoded as a string in the following format:
+	// `<policyPackName>/<policyPackVersion>/<policyName>`
+	// For example, "aws-security/5/no-public-s3-buckets"
+	Identifier       string           `json:"identifier"`
+	EnforcementLevel EnforcementLevel `json:"enforcementLevel"`
```

Comment:
Just so I have an idea of where we're headed: the server will have information about things like `EnforcementLevel` already -- what happens if this information is different? Also, is the plan to put all the fields in the policy model into this struct eventually, or only some of them, like `EnforcementLevel` (and what rationale separates these two buckets)?

Reply:
I would hope the info is not different bc it would be mean something has gone terribly wrong.

By putting enforcement level here, I wont have to join these events with the `Policy`. Basically all the information to display this will be self contained in the event.